tools/gltfViewer/main.cpp

```diff
@@ -0,0 +1,181 @@
+    : QGuiApplication(argc, argv, aflags)
+    , m_context(nullptr)
+{
+}
+
+ViewerApplication::~ViewerApplication() = default;
+
+bool ViewerApplication::event(QEvent *event)
+{
+    if (event->type() == QEvent::FileOpen) {
+        QFileOpenEvent *fileEvent = static_cast<QFileOpenEvent *>(event);
+        if (m_context)
+            m_context->setContextProperty("_gltfFile", fileEvent->url());
+    }
+
+    return QGuiApplication::event(event);
+}
+
+bool ViewerApplication::eventFilter(QObject *watched, QEvent *event)
+{
+    if (event->type() == QEvent::Drop) {
+        QDropEvent *de = static_cast<QDropEvent *>(event);
+        if (de->mimeData()->urls().size() == 1) {
+            QUrl u = de->mimeData()->urls().front();
+            if (m_context)
+                m_context->setContextProperty("_gltfFile", u);
+            de->accept();
+            return false;
+        }
+    } else if (event->type() == QEvent::DragEnter) {
+        QDragEnterEvent *de = static_cast<QDragEnterEvent *>(event);
+        if (de->mimeData()->urls().size() == 1) {
+            QUrl u = de->mimeData()->urls().front();
+            if (u.scheme() == QLatin1String("file")) {
+                de->accept();
+                return false;
+            }
+        }
+    } else if (event->type() == QEvent::Resize) {
+        QResizeEvent *re = static_cast<QResizeEvent *>(event);
+        if (!re->size().isEmpty() && m_context)
+            m_context->setContextProperty("_winSize", re->size());
+    }
+
+    return QGuiApplication::eventFilter(watched, event);
+}
+
+void ViewerApplication::viewAll(Qt3DRender::QCamera *camera)
+{
+    if (!camera)
+        return;
+
+    connect(camera, &Qt3DRender::QCamera::viewMatrixChanged, this, [camera, this]() {
+        const QVector3D position = camera->position();
+        const QVector3D viewCenter = camera->viewCenter();
+        const float sceneExtent = 2.0f * (viewCenter - position).length();
+        camera->setFarPlane(10.0f * sceneExtent); // add some headroom to be able to zoom out
+        camera->setNearPlane(0.001f * sceneExtent);
+        camera->disconnect(this);
+    });
+    camera->viewAll();
+}
+
+int main(int argc, char *argv[])
+{
+    QCoreApplication::setApplicationName("gltfViewer");
+    QCoreApplication::setOrganizationDomain("kdab.com");
+    QCoreApplication::setOrganizationName("KDAB");
+    QCoreApplication::setApplicationVersion(QStringLiteral("1.0"));
+
+    qmlRegisterType<OrbitCameraController>("KuesaViewer", 1, 0, "OrbitCameraController");
+
+    ViewerApplication app(argc, argv);
+
+    QCommandLineParser cmdline;
+    cmdline.addHelpOption();
+    cmdline.addVersionOption();
+    cmdline.setApplicationDescription(
+            QObject::tr("\nThis tool renders glTF 2.0 files.\n"
+                        "Example : ./gltfViewer -f my_file.gltf"));
+    cmdline.addPositionalArgument("source", QCoreApplication::translate("main", "gltf 2.0 file to render."));
```

Comment:
isnt --input (-i) more natural?

Reply:
-f is for full screen. Guess I should mark it as optional. File is actually a positional argument